nemo/utils/model_utils.py

```diff
@@ -0,0 +1,232 @@
+
+    if 'manifest' in filename:
+        filename = filename.replace('manifest', '')
+
+    if '_' != filename[-1]:
+        filename = filename + '_'
+
+    return filename
+
+
+def resolve_validation_dataloaders(model: 'ModelPT'):
+    """
+    Helper method that operates on the ModelPT class to automatically support
+    multiple dataloaders for the validation set.
+
+    It does so by first resolving the path to one/more data files via `resolve_filepath_from_cfg()`.
+    If this resolution fails, it assumes the data loader is prepared to manually support / not support
+    multiple data loaders and simply calls the appropriate setup method.
+
+    If resolution succeeds:
+        Checks if provided path is to a single file or a list of files.
+        If a single file is provided, simply tags that file as such and loads it via the setup method.
+        If multiple files are provided:
+            Inject a new manifest path at index "i" into the resolved key.
+            Calls the appropriate setup method to set the data loader.
+            Collects the initialized data loader in a list and preserves it.
+            Once all data loaders are processed, assigns the list of loaded loaders to the ModelPT.
+            Finally assigns a list of unique names resolved from the file paths to the ModelPT.
+
+    Args:
+        model: ModelPT subclass, which requires >=1 Validation Dataloaders to be setup.
+    """
+    cfg = copy.deepcopy(model._cfg)
+    dataloaders = []
+
+    # process val_loss_idx
+    if hasattr(cfg.validation_ds, 'val_loss_idx'):
+        cfg = OmegaConf.to_container(cfg)
+        val_loss_idx = cfg['validation_ds'].pop('val_loss_idx')
+        cfg = OmegaConf.create(cfg)
+    else:
+        val_loss_idx = 0
+
+    # Set val_loss_idx
+    model._validation_loss_idx = val_loss_idx
+
+    filepath_key = resolve_filepath_from_cfg(cfg.validation_ds)
+
+    if filepath_key is None:
+        logging.debug(
+            "Could not resolve file path from provided config - {}. "
+            "Disabling support for multi-dataloaders.".format(cfg.validation_ds)
+        )
+
+        model.setup_validation_data(cfg.validation_ds)
+        model._validation_filenames = ["validation_"]
+        return
+
+    manifest_paths = cfg.validation_ds[filepath_key]
+
+    if type(manifest_paths) in (list, tuple, ListConfig):
+
+        for filepath_val in manifest_paths:
+            cfg.validation_ds[filepath_key] = filepath_val
+            model.setup_validation_data(cfg.validation_ds)
+            dataloaders.append(model._validation_dl)
+
+        model._validation_dl = dataloaders
+        model._validation_filenames = [parse_filepath_as_name(fp) for fp in manifest_paths]
+
+        # In fast-dev-run, only one data loader is used
+        if model._trainer.fast_dev_run:
+            model._validation_dl = model._validation_dl[:1]
+            model._validation_filenames = model._validation_filenames[:1]
```

Comment:
[COMMENT]

Assumption that there is some kind of a `manifest_path` and all datasets have it simply doesn't hold for CV. Datasets - and tasks down there are much more diverse.

I also think it also doesn't hold for NLP - but ask the NLP experts for their comments.

Reply:
As you have said correctly, I make the assumption that there exists at least one file path (dosnt need to be named `manifest_filepath`, but the code also backs off entirely if thats not the case. 

If it finds no such manifest or test filepath, it will go fall into the test above before this if-else block and trust the model developer will handle this case. 

If the entire config doesnt exist (say for CV there is no validation_ds or test_ds path), ModelPT simply will never call this method for resolution. 

As such, nothing breaks. This can be noted that all tests (including NLP and CV pass).